packages/gallery/src/lib/dom.js

```diff
@@ -0,0 +1,102 @@
+import { sanitise, updateCSS } from './utils';
+import { ATTRIBUTE, EVENTS, KEYCODES } from './constants';
+
+export const init = store => () => {
+    const state = store.getState();
+    const { settings, dom, list } = state;
+
+    if (!dom.liveRegion) console.warn(`A live region announcing current and total items is recommended for screen readers.`);
+    else writeLiveRegion(state);
+    /* istanbul ignore next */
+    if (dom.fullscreen) {
+        if (document.fullscreenEnabled || document.webkitFullscreenEnabled) {
+            dom.fullscreen.addEventListener('click', toggleFullScreen.bind(null, store));
+            document.addEventListener('fullscreenchange', e => {
+                if (document.fullscreenElement) document.documentElement.classList.add(settings.className.fullscreen);
+                else document.documentElement.classList.remove(settings.className.fullscreen);
+            });
+        } else dom.fullscreen.parentNode.removeChild(dom.fullscreen);
+    }
+    if (dom.previous) dom.previous.addEventListener('click', previous.bind(null, store));
+    if (dom.next) dom.next.addEventListener('click', next.bind(null, store));
+    if (dom.triggers.length) {
+        dom.triggers.forEach(trigger => trigger.addEventListener('click', () => {
+            goTo(store)(+trigger.getAttribute(ATTRIBUTE.NAVIGATE));
+        }));
+    }
+
+    /* istanbul ignore next */
+    list.addEventListener('keydown', e => {
+        switch (e.keyCode) {
+        case KEYCODES.LEFT:
+            e.preventDefault();
+            previous(store);
+            break;
+        case KEYCODES.RIGHT:
+            e.preventDefault();
+            next(store);
+            break;
+        // case KEYCODES.TAB:
+        //     next(store);
+        //     break;
+        }
+    });
+
+    updateCSS(state);
+    broadcast(store, EVENTS.INITIALISED)(state);
+};
+
+const writeLiveRegion = ({ activeIndex, items, settings, dom }) => dom.liveRegion.innerHTML = sanitise(settings.announcement(activeIndex + 1, items.length));

```

Comment:
Would it be taking things too far to have this conditionally done based on a setting?  The reason I ask is I was experimenting with the CSS, and found that things still worked really quite nicely - even when changing the grid-auto-columns to 50%, or 33% - for more slider like behaviour.  The only thing that didn't work was tracking the active card.

At that point, if more than one is visible - an active slide doesn't really make sense, but removing the live region from the markup only errored.  Might be making things more complicated than is needed though!